src/engine/MaxConstraint.cpp

```diff
@@ -408,28 +452,94 @@ List<PiecewiseLinearCaseSplit> MaxConstraint::getCaseSplits() const
         splits.append( getSplit( _f ) );
     }
 
+    if ( _eliminatedVariables ) { // todo check with GuyK (I didn't add tightenings)
+        PiecewiseLinearCaseSplit phaseOfEliminatedIsMax;
+        // TODO - GuyA - move up to the case when _f exists only?
+        // f = maxEliminated
+        Equation maxEquation(Equation::EQ);
+        maxEquation.setScalar(_maxValueOfEliminated);
```

Comment:
no need to add an equation, just tighten f's lower and upper bounds to _maxValueOfEliminated

Reply:
done